CMakeLists.txt

```diff
@@ -23,9 +23,16 @@ set(CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/fortran )
 
 install(DIRECTORY ${CMAKE_BINARY_DIR}/fortran/ DESTINATION include/fortran FILES_MATCHING PATTERN "*.mod" PATTERN "CMake*" EXCLUDE )
 
-set(CXX11_FLAGS "-std=c++11")
-
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+# Based on http://stackoverflow.com/a/31010221/592024
+if (CMAKE_VERSION VERSION_LESS "3.1")
+    if("${CMAKE_Fortran_COMPILER}" MATCHES "ftn")
```

Comment:
Wait a second, are you sure this works on CSCS machines? `ftn` is the fortran compiler wrapper, so if you have the `prgEnv-gnu` loaded `ftn` will represent the GNU compiler.

I think you need something more sophisticated to take the decision here.